mino/minows/address_test.go

```diff
@@ -9,65 +9,78 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-const (
-	AddrAllInterface = "/ip4/0.0.0.0/tcp/80"
-	AddrLocalhost    = "/ip4/127.0.0.1/tcp/80"
-	AddrHostname     = "/dns4/example.com/tcp/80"
-)
-const (
-	PID1 = "QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU"
-	PID2 = "QmVt9t5Tk2uEoA4CDbKNCVxqrut8UXmWHXvFZ8wFZ3ghhv"
-)
-
-func TestNewAddress(t *testing.T) {
+func Test_newAddress(t *testing.T) {
+	const pid1 = "QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU"
+	const pid2 = "QmVt9t5Tk2uEoA4CDbKNCVxqrut8UXmWHXvFZ8wFZ3ghhv"
+	const addrAllInterface = "/ip4/0.0.0.0/tcp/80"
+	const addrLocalhost = "/ip4/127.0.0.1/tcp/80"
+	const addrHostname = "/dns4/example.com/tcp/80"
+	type args struct {
+		location string
+		identity string
+	}
 	tests := map[string]struct {
-		location ma.Multiaddr
-		identity peer.ID
-		err      bool
+		args args
 	}{
-		"missing location": {
-			location: nil,
-			identity: mustCreatePeerID(PID1),
-			err:      true,
-		},
-		"missing identity": {
-			location: mustCreateMultiaddress(AddrAllInterface),
-			identity: "",
-			err:      true,
-		},
 		"all interface": {
-			location: mustCreateMultiaddress(AddrAllInterface),
-			identity: mustCreatePeerID(PID1),
+			args: args{addrAllInterface, pid1},
 		},
 		"localhost": {
-			location: mustCreateMultiaddress(AddrLocalhost),
-			identity: mustCreatePeerID(
-				PID2),
+			args: args{addrLocalhost, pid2},
 		},
 		"hostname": {
-			location: mustCreateMultiaddress(AddrHostname),
-			identity: mustCreatePeerID(
-				PID1),
+			args: args{addrHostname, pid1},
 		},
 	}
+	t.Parallel() // run this test function args parallel to other test functions
+	for name, tt := range tests {
+		tt := tt // capture range variable
+		t.Run(name, func(t *testing.T) {
+			t.Parallel() // run this test case args parallel to other test cases
+			location := mustCreateMultiaddress(t, tt.args.location)
+			identity := mustCreatePeerID(t, tt.args.identity)
 
+			// no exported field on 'address' type, ignored
+			_, err := newAddress(location, identity)
+			require.NoError(t, err)
+		})
+	}
+}
+
+func Test_newAddress_Invalid(t *testing.T) {
+	const pid1 = "QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU"
+	const addrAllInterface = "/ip4/0.0.0.0/tcp/80"
+
+	tests := map[string]struct {
+		location ma.Multiaddr
+		identity peer.ID
+	}{
+		"missing location": {
+			location: nil,
+			identity: mustCreatePeerID(t, pid1),
+		},
+		"missing identity": {
+			location: mustCreateMultiaddress(t, addrAllInterface),
+			identity: "",
+		}}
 	t.Parallel() // run this test function in parallel to other test functions
```

Comment:
Aren't all tests run in parallel anyway? Except the ones where you call `t.Run`, where `t.Parallel` makes sense.

Reply:
https://go.dev/wiki/TableDrivenTests#parallel-testing

Reply:
You're correct. It's tests from different packages which are run in parallel. See my comment above.